src/dataset-builder/CohortEditor.test.ts

```diff
@@ -93,23 +93,29 @@ describe('CohortEditor', () => {
       })
     );
 
+  beforeEach(() => {
+    mockDataRepo([getSnapshotBuilderCountMock()]);
+  });
+
   it('renders unknown criteria', async () => {
     // Arrange
-    mockDataRepo([getSnapshotBuilderCountMock(0)]);
     const criteria = { name: 'bogus', invalid: 'property' };
+    let consoleError = '';
     // This should error fetching the count because the conversion to API counts to generate counts should fail
-    jest.spyOn(console, 'error').mockImplementation((error) => expect(error).toBe('Error fetching count for criteria'));
```

Comment:
Did you figure out why this approach didn't work?

Reply:
Not really, my guess is that the assert inside the lambda fails but since it's not at the top level, the failure is hidden. Once I got it working by capturing the output and then asserting at the end of the test I stopped looking into it.